packages/webapp/src/routes/credits-purchase/credits-metric-upload.js

```diff
@@ -0,0 +1,137 @@
+import { deleteBlobFromContainers } from '../../utils/azure-storage.js'
+import { buildConfig } from '../../utils/build-upload-config.js'
+import { uploadFile } from '../../utils/upload.js'
+import { getMaximumFileSizeExceededView, getMetricFileValidationErrors } from '../../utils/helpers.js'
+import { ThreatScreeningError, MalwareDetectedError } from '@defra/bng-errors-lib'
+import creditsPurchaseConstants from '../../utils/credits-purchase-constants.js'
+
+const UPLOAD_CREDIT_METRIC_ID = '#uploadMetric'
+
+const processSuccessfulCreditUpload = async (result, request, h) => {
+  const creditsValidationError = getMetricFileValidationErrors(result.postProcess.metricData?.validation)
+  if (creditsValidationError) {
+    await deleteBlobFromContainers(result.config.blobConfig.blobName)
+    return h.view(creditsPurchaseConstants.views.CREDITS_PURCHASE_UPLOAD_METRIC, creditsValidationError)
+  }
+
+  request.yar.set(creditsPurchaseConstants.redisKeys.CREDITS_PURCHASE_METRIC_LOCATION, result.config.blobConfig.blobName)
+  request.yar.set(creditsPurchaseConstants.redisKeys.CREDITS_PURCHASE_METRIC_FILE_SIZE, result.fileSize)
+  request.yar.set(creditsPurchaseConstants.redisKeys.CREDITS_PURCHASE_METRIC_FILE_TYPE, result.fileType)
+  request.yar.set(creditsPurchaseConstants.redisKeys.CREDITS_PURCHASE_METRIC_DATA, result.postProcess.metricData)
+  request.yar.set(creditsPurchaseConstants.redisKeys.CREDITS_PURCHASE_METRIC_FILE_NAME, result.filename)
+  request.logger.info(`${new Date().toUTCString()} Received metric data for ${result.config.blobConfig.blobName.substring(result.config.blobConfig.blobName.lastIndexOf('/') + 1)}`)
+  return h.redirect(creditsPurchaseConstants.routes.CREDITS_PURCHASE_CHECK_UPLOAD_METRIC)
+}
+
+const processCreditsErrorUpload = (err, h) => {
+  switch (err.message) {
+    case creditsPurchaseConstants.uploadErrors.emptyFile:
+      return h.view(creditsPurchaseConstants.views.CREDITS_PURCHASE_UPLOAD_METRIC, {
+        err: [{
+          text: 'The selected file is empty',
+          href: UPLOAD_CREDIT_METRIC_ID
+        }]
+      })
+    case creditsPurchaseConstants.uploadErrors.noFile:
+      return h.view(creditsPurchaseConstants.views.CREDITS_PURCHASE_UPLOAD_METRIC, {
+        err: [{
+          text: 'Select a Biodiversity Metric',
```

Comment:
Error message should be `Upload the statutory biodiversity metric file with the ‘unit shortfall summary’ tab` to match the copydeck